sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -428,17 +468,25 @@ public PDone expand(PCollection<T> input) {
     @Override
     public void populateDisplayData(DisplayData.Builder builder) {
       super.populateDisplayData(builder);
+      checkState(
+          getFilenamePrefix() != null,
+          "Unable to populate DisplayData for invalid AvroIO.Write (unset output prefix).");
+      String outputPrefixString = null;
+      if (getFilenamePrefix().isAccessible()) {
```

Comment:
We do have some support: https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java#L875

I'm not sure whether that overload would be sufficient here as it needs to infer the value type for formatting. I'd be happy to see `DisplayData` APIs evolve to better support `ValueProvider<T>`. If DisplayData is hard to use, transform authors won't add it.